### ISSUE TITLE
Removed `std::r#await` macro causes compilation fail

### ISSUE THREAD

**Issue by Empty2k12** [tokio-rs/tokio#1087]
## Version

```
❯ cargo tree | rg tokio
    Updating crates.io index
├── tokio v0.1.19
│   ├── tokio-async-await v0.1.7
│   │   └── tokio-io v0.1.12
│   ├── tokio-codec v0.1.1
│   │   └── tokio-io v0.1.12 (*)
│   ├── tokio-current-thread v0.1.6
│   │   └── tokio-executor v0.1.7
│   ├── tokio-executor v0.1.7 (*)
│   ├── tokio-fs v0.1.6
│   │   ├── tokio-io v0.1.12 (*)
│   │   └── tokio-threadpool v0.1.14
│   │       └── tokio-executor v0.1.7 (*)
│   │   └── tokio-io v0.1.12 (*)
│   ├── tokio-io v0.1.12 (*)
│   ├── tokio-reactor v0.1.9
│   │   ├── tokio-executor v0.1.7 (*)
│   │   ├── tokio-io v0.1.12 (*)
│   │   └── tokio-sync v0.1.5
│   ├── tokio-sync v0.1.5 (*)
│   ├── tokio-tcp v0.1.3
│   │   ├── tokio-io v0.1.12 (*)
│   │   └── tokio-reactor v0.1.9 (*)
│   ├── tokio-threadpool v0.1.14 (*)
│   ├── tokio-timer v0.2.10
│   │   └── tokio-executor v0.1.7 (*)
│   ├── tokio-trace-core v0.1.0
│   ├── tokio-udp v0.1.3
│   │   ├── tokio-codec v0.1.1 (*)
│   │   ├── tokio-io v0.1.12 (*)
│   │   └── tokio-reactor v0.1.9 (*)
│   └── tokio-uds v0.2.5
│       ├── tokio-codec v0.1.1 (*)
│       ├── tokio-io v0.1.12 (*)
│       └── tokio-reactor v0.1.9 (*)
    ├── tokio v0.1.19 (*)
    └── tokio-async-await v0.1.7 (*)
```

## Platform

`Darwin 18.6.0 Darwin Kernel Version 18.6.0: Thu Apr 25 23:49:07 PDT 2019; root:xnu-4903.261.4~4/RELEASE_X86_64 x86_64`

## Subcrates

tokio-futures

## Description

Running with latest nightly, the compilation of `tokio-async-await` fails, as the `std::r#await` macro which is being rexported has been deleted in this commit: https://github.com/rust-lang/rust/pull/60675.

I expected to see this happen:

```
   Compiling my-bin v0.1.0 (/Users/gero/dev/my-bin)
    Finished dev [unoptimized + debuginfo] target(s) in 4.29s
     Running `target/debug/my-bin`
```

Instead, this happened: 
```
   Compiling tokio-async-await v0.1.7
error[E0432]: unresolved import `std::await`
  --> /Users/gero/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-async-await-0.1.7/src/lib.rs:35:9
   |
35 | pub use std::await as std_await;
   |         ^^^^^^^^^^^^^^^^^^^^^^^ no `await` in the root
```
-->

**Comment by sdroege** [comment]
I locally worked around this with the following change. It's enough for getting my code to run (using the futures 0.3 -> 0.1 conversions), but the async/await examples in tokio are failing to compile. That's why I didn't send a PR.

```diff
diff --git a/tokio-futures/src/await.rs b/tokio-futures/src/await.rs
index 1e8f6e7..c357539 100644
--- a/tokio-futures/src/await.rs
+++ b/tokio-futures/src/await.rs
@@ -1,16 +1,15 @@
 /// Wait for a future to complete.
 #[macro_export]
-macro_rules! await {
+macro_rules! r#await {
     ($e:expr) => {{
         #[allow(unused_imports)]
         use $crate::compat::backward::IntoAwaitable as IntoAwaitableBackward;
         #[allow(unused_imports)]
         use $crate::compat::forward::IntoAwaitable as IntoAwaitableForward;
-        use $crate::std_await;
 
         #[allow(unused_mut)]
         let mut e = $e;
         let e = e.into_awaitable();
-        std_await!(e)
+        e.await
     }};
 }
diff --git a/tokio-futures/src/lib.rs b/tokio-futures/src/lib.rs
index 411dc42..fdf709d 100644
--- a/tokio-futures/src/lib.rs
+++ b/tokio-futures/src/lib.rs
@@ -28,8 +28,3 @@ pub mod compat;
 pub mod io;
 pub mod sink;
 pub mod stream;
-
-// Rename the `await` macro in `std`. This is used by the redefined
-// `await` macro in this crate.
-#[doc(hidden)]
-pub use std::await as std_await;
diff --git a/tokio/src/lib.rs b/tokio/src/lib.rs
index 872f23c..8b89a83 100644
--- a/tokio/src/lib.rs
+++ b/tokio/src/lib.rs
@@ -1,5 +1,5 @@
 #![doc(html_root_url = "https://docs.rs/tokio/0.1.19")]
-#![deny(missing_docs, warnings, missing_debug_implementations)]
+#![deny(missing_docs, missing_debug_implementations)]
 #![cfg_attr(feature = "async-await-preview", feature(async_await, await_macro))]
 
 //! A runtime for writing reliable, asynchronous, and slim applications.
```

**Comment by taiki-e** [comment]
See also https://github.com/tokio-rs/tokio/pull/1080 and https://github.com/tokio-rs/tokio/pull/1082.

**Comment by carllerche** [comment]
Thanks for the report. Progress is ongoing to address the issue (see above).

However, the fixes will not be applied to the 0.1 branch. For the immediate future, to track the latest async / await changes with Tokio, using the version off of the git master branch will be required.

**Comment by sdroege** [comment]
For the time being, https://rust-lang-nursery.github.io/futures-api-docs/0.3.0-alpha.16/futures/compat/index.html might be useful then, and removing the translation layer from tokio 0.1 to prevent confusion.
